### PR TITLE
Add support for ZooKeeper server 3.6.2

### PR DESCRIPTION
**Pull request by hmusum**
Will merge when https://github.com/vespa-engine/vespa/pull/15727 has been through all tests

### CODE REVIEW

File: zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/ZooKeeperServer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.apache.zookeeper.server.quorum.QuorumPeerMain;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Class to start zookeeper server. Extends QuorumPeerMain to be able to call initializeAndRun() and wraps
+ * exceptions so it can be used by code that does not depend on zookeeper.
+ */
+class ZooKeeperServer extends QuorumPeerMain {
```
Comment (mpolden) [inline zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/ZooKeeperServer.java:15]: Missing `@author`.

File: zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/VespaZooKeeperAdminImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.admin.ZooKeeperAdmin;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@SuppressWarnings("unused") // Created by injection
+public class VespaZooKeeperAdminImpl implements VespaZooKeeperAdmin {
```
Comment (mpolden) [inline zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/VespaZooKeeperAdminImpl.java:13]: Missing `@author`.

**Review by mpolden — APPROVED** [review]

File: zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/ZooKeeperServer.java (hunk 1 above)
Comment (hmusum) [reply zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/ZooKeeperServer.java:17]: Fixed

File: zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/VespaZooKeeperAdminImpl.java (hunk 2 above)
Comment (hmusum) [reply zookeeper-server/zookeeper-server-3.6.2/src/main/java/com/yahoo/vespa/zookeeper/VespaZooKeeperAdminImpl.java:16]: Fixed

**Review by mpolden — APPROVED** [review]

**Pull request merged**